From 2aacf2d696b447eecd2353df2084436e6b47dbfe Mon Sep 17 00:00:00 2001
From: Jordan Paige Hendricks <jordan.hendricks@joyent.com>
Date: Fri, 20 Jul 2018 21:12:14 +0000
Subject: [PATCH] joyent/node-fast#20 allow specifying ports used for fastbench
 CLI Reviewed by: David Pacheco <dap@joyent.com> Approved by: David Pacheco
 <dap@joyent.com>

---
 bin/fastbench | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/bin/fastbench b/bin/fastbench
index 644fedf..9d54ab6 100755
--- a/bin/fastbench
+++ b/bin/fastbench
@@ -74,8 +74,10 @@ function fbUsageMessage()
 		fbDflConcurrency + ')',
 	    '    -d | --duration NSECONDS     Stop after NSECONDS seconds.',
 	    '    -i | --interval NSECONDS     Report every NSECONDS seconds.',
+	    '    -k | --kang-port PORT         Use PORT for kang endpoint.',
 	    '    -n | --nrequests NREQUESTS   Stop after completing ' +
 		'NREQUESTS requests.',
+	    '    -p | --artedi-port PORT       Use PORT for metrics endpoint.',
 	    ''
 	].join('\n');
 
@@ -106,7 +108,9 @@ function main()
 	});
 
 	parser = new mod_getopt.BasicParser(
-	    'c:(count)d:(duration)i:(interval)n:(nrequests)', process.argv);
+	    'c:(count)d:(duration)i:(interval)n:(nrequests)' +
+		'p:(artedi-port)k:(kang-port)',
+	     process.argv);
 	while ((option = parser.getopt()) !== undefined) {
 		switch (option.option) {
 		case 'c':
@@ -126,6 +130,14 @@ function main()
 			config.fbNRequests = parseOptInt(option, 1);
 			break;
 
+		case 'p':
+			config.fbArtediPort = parseOptInt(option, 1);
+			break;
+
+		case 'k':
+			config.fbKangPort = parseOptInt(option, 1);
+			break;
+
 		default:
 			/* error message already emitted */
 			mod_assertplus.equal('?', option.option);
-- 
2.21.0

